### PR TITLE
Additions

### PR DESCRIPTION
**Pull request by knil-maloon**
- Added more explanations on the info alert.
- The new variable name gets checked for special characters, or if it begins on a number, etc. If added wrongly it displays a warning message.
- Added check to ensure primary key is not "editable".
- Rearranged Autocomplete function on smaller functions. #ToDo on converting data into objects.

### CODE REVIEW

File: REDCapInstrumentAndVariableSQLRenamer.php [hunk 1]
```diff
@@ -43,7 +43,7 @@ public function printVariableList($pid): string
                 $aux = $row['form_name'];
                 $list_html .= "<div class='group-header'>" . REDCap::getInstrumentNames($row['form_name']) . "</div>";
             }
-            if ($row['field_name'] != $row['form_name'] . "_complete" && $row['field_name'] != "record_id") {
+            if ($row['field_name'] != $row['form_name'] . "_complete" && $row['field_order'] != "1") {
```
Comment (paulj3) [inline REDCapInstrumentAndVariableSQLRenamer.php:46]: @knil-maloon is this the best way to check for the primary key? there isn't another identifier on the table? we just have to see if it is the first one?

File: classes/Autocomplete.php [hunk 2]
```diff
@@ -147,18 +158,31 @@ public static function getAutocompleteData($module, $pid, $getTerm, $type, $opti
...
+    {
+        $search_term = self::sanatizingSearchTerms($getTerm, $type, $option);
+
+        $sqlData = self::executeSQLSearch($module, $pid, $type, $option, $search_term);
+        $q = $sqlData[0];
+        $subtype = $sqlData[1];
+        $this_term = $sqlData[2];
+        $search_terms = $sqlData[3];
+
+        return self::processResults($q, $option, $type, $this_term, $search_terms, $subtype);
```
Comment (paulj3) [inline classes/Autocomplete.php:186]: This is beautiful...poetic, even! well done!

File: REDCapInstrumentAndVariableSQLRenamer.php (hunk 1 above)
Comment (knil-maloon) [reply REDCapInstrumentAndVariableSQLRenamer.php:46]: Yup! I asked Rob to make sure and that is what he said, the first field is always the primary key.

Comment (paulj3) [reply REDCapInstrumentAndVariableSQLRenamer.php:46]: okay awesome, thank you for doing that.

**Review by paulj3 — APPROVED** [review]

**Pull request merged**